workflows/utils/prepS2.py

```diff
@@ -326,6 +328,73 @@ def s2_ndvi(red_file, nir_file, out_file=False):
 
     return ndvi
 
+def scale_sentinel2_l2a(cog_dir, scale_dir, new_dtype='float32'):
+    """
+    Apply scale factor to Sentinel Level-2A data
+    (Quantification value of 10,000 and no offset value)
+    https://docs.sentinel-hub.com/api/latest/data/sentinel-2-l2a/
+
+    Note: Check on assigning new data type of float32 and converting to COG multiple times.
+    """
+    # All the files in the directory (includes QA bands and xml file)
+    filenames = glob.glob(f"{cog_dir}/*")
+
+    for f in filenames:
+        f_name = f.split('/')[-1]
+        out_path = f"{scale_dir}{f_name}"
+
+        # Split the file name to figure out the product
+        file_parts = f_name.split('_')
+        prod_name = f"{file_parts[-2]}_{file_parts[-1][:-4]}".lower()
+
+        # Apply scaling for surface reflectance bands
+        if f[-11:].startswith('B') and f[-11:].endswith('.tif'):
+            quant_value = 10000
+            boa_offset = 0
+            nodata = 0
+            apply_scale_factor_sentinel(f, quant_value, boa_offset, nodata, out_path, new_dtype)
+            logging.info(f"Prod name {prod_name} scaled with scale factor {quant_value}, offset {boa_offset}")
+        
+        # Apply scaling for AOT band
+        elif f[-11:].startswith('AOT'):
+            quant_value = 1000
+            boa_offset = 0
+            nodata = 0
+            apply_scale_factor_sentinel(f, quant_value, boa_offset, nodata, out_path, new_dtype)
+            logging.info(f"Prod name {prod_name} scaled with scale factor {quant_value}, offset {boa_offset}")
+        
+        # For the remaining files (QA band, xml, etc.), just copy into new directory
+        else:
+            shutil.copy(f, out_path)
+
+def apply_scale_factor_sentinel(input_data, quant_value, boa_offset, nodata, out_path, new_dtype='float32'):
+        """
+        Apply the scale factor to a tif of a sentinel product.
+        """
+    
+        # Open the tif as xarray to apply scale factor + offset
+        logging.info(f"Opening {input_data} as xarray")
+        img_arr = rxr.open_rasterio(input_data)
+        logging.info(f"MIN, MAX BEFORE {img_arr.min()}, {img_arr.max()}")
+        
+        # Nodata mask (nodata value of 0 for Sentinel 2)
+        mask = img_arr.data == nodata
+
+        # Apply the scale factor and offset
+        img_arr_rescaled = img_arr / quant_value + boa_offset
+
+        # New datatype of float32 because new values are often less than 1 
+        img_arr_rescaled = img_arr_rescaled.astype(new_dtype)
+
+        # Making sure any original nodata is still nodata (I think?)
+        img_arr_rescaled.data[mask] = nodata
+
+        logging.info(f"MIN, MAX AFTER {img_arr_rescaled.min()}, {img_arr_rescaled.max()}")
+        logging.info(f"Data-type AFTER: {img_arr_rescaled.dtype}")
+
+        # Convert xarray to COG 
+        img_arr_rescaled.rio.to_raster(raster_path = out_path, driver="COG")
```

Comment:
does this retain CRS and other metadata?

Reply:
uhh havent checked, @sabinenix ?

Reply:
It looks to me like the outputs are still located correctly relative to the originals and I think the metadata itself is copied from the original download directory so nothing should be lost there. With the landsat data though they get re-cogged after scaling using the original code's cog function - do you think that order would that be preferable for sentinel too?

Reply:
I think that re-cogging is a good idea for sentinel